### DISCUSSION TITLE
Ice with Rust

### DISCUSSION THREAD

**Discussion by dgehriger** [zeroc-ice/ice#2317] [Q&A] [answered]
Hi,

I'm considering using ice for our mixed C++ / Rust environment. While there is a crate supporting Rust (ice-rs), it has gone stale three years ago. Also, it's GPL. On the other hands, I'm seeing signs of Rust support being implemented by the zeroc-ice project. Would you mind telling me what the current status is, and if it is already possible to use ZeroC-Ice from Rust?

**Comment by bernardnormier** [comment, marked as answer]
Hi Daniel,

We very much welcome independent efforts to bring Ice to more programming languages. `ice-rs` was such an effort, but unfortunately it looks like it went stale a few years ago.

I am not aware of any other "Ice for Rust" project. We don't plan to add support for Rust in Ice--there are portions of the Slice language (in Ice) that don't map well to Rust, namely classes and exceptions.

On the other hand, we very much plan to implement [IceRPC](https://github.com/icerpc/) in Rust. That's the next language we'll support in IceRPC. And we were careful to design ["new Slice"](https://docs.icerpc.dev/slice2) to work well with Rust. We even wrote the new Slice compiler (`slicec`) in Rust.

IceRPC for Rust is not public yet. I hope we can release it in the coming months, but there is no set timeframe at this point.

**Comment by InsertCreativityHere** [comment]
Just to expand and clarify here,
[IceRPC](https://zeroc.com/icerpc) and [Ice](https://zeroc.com/ice) are both developed by us, and so have a high degree of interop.
Once we release Rust support with IceRPC, it would be very possible to use your existing Ice C++ code alongside IceRPC Rust code!

**Reply by dgehriger** [reply]
Thank you very much for your complete and fast response! I'll keep your project on my radar then. I'm currently writing a few benchmarks that nicely represent my specific use cases, and I may well try to port them to Ice to get some hard data. Thanks again.
